src/runtime/internal/static.ts

```diff
@@ -49,7 +49,11 @@ export default eventHandler((event) => {
     "",
   ];
   if (encodings.length > 1) {
-    setResponseHeader(event, "Vary", "Accept-Encoding");
```

Comment:
Maybe we can try `appendResponseHeader`?

Reply:
I _knew_ there was a better way but couldn't find it :D

![image](https://github.com/user-attachments/assets/1d652b2b-2cf0-4d9c-9b74-6a51ecbe5539)

Still curious why it works different all the time... 

Anyways, fixed.